src/api.js

```diff
@@ -2,15 +2,22 @@ import { examRequiresAccessToken, store } from './data';
 
 export function isExam() {
   const { exam } = store.getState().examState;
-  return exam.id !== null;
+  return !!exam?.id;
 }
 
 export function getExamAccess() {
-  const { examAccessToken } = store.getState().examState;
+  const { exam, examAccessToken } = store.getState().examState;
+  if (!exam) {
+    return '';
+  }
   return examAccessToken.exam_access_token;
 }
 
 export async function fetchExamAccess() {
+  const { exam } = store.getState().examState;
   const { dispatch } = store;
+  if (!exam) {
+    return Promise.resolve();
```

Comment:
Note: I had it return a resolved promise. When I was playing with promises, I noticed that even if I did Promise.reject(), I would need to catch the error that is thrown and the promise returned would essentially return a resolved Promise (which was why I had the whole block with the catch error included before). Also if we're not logging an error anyway, then the error part is probably extraneous anyway. Lemme know what you think.

The frontend app learning shouldn't really get here, but even if the exam access token isn't granted, I have a check to only unblock exam access for exams if there is an exam access token (so not allowing for blank exam access tokens).